TelosLabs/giving-connection
# Feature | Map improvements card

### Context
A popup with more and detailed information of an organization was needed to appear inside the map

### Some important _spelling background_ before review:

- **Pin/Marker:**
 Refers to the blue map icons displaying all over the map.
- **Larger Popup:** 
Refers to the main change of this feature ⭐ , the larger card that is open on left side of the map. With extra information.
- **Smaller Popup:** 
Refers to the common info window shown under the selected pin.



### What changed

1. Javascript Controller was added, adding specific behavior when a location/pin is clicked or hovered.


      - When either a pin or left-side panel locations are clicked, a larger popup is visible for the user.
      - When hover a pin, the smaller popup appears under the pin.
    


      - When a pin is clicked, the smaller popup appears under the pin.
2. Session Storage manages selected pins, smaller and larger map popups.
3. Pins/Markers Change color when clicked.

### How to test it

1. `rails db:seed`
2. `yarn install`
3. `bundle install`
4. `hivemind`
5. Test the behavior in desktop view by clicking different pins on the map. (You should be able to see and interact with larger popup)
6. Test the behavior in mobile view by clicking different pins on the map.

### References

[Notion ticket](https://www.notion.so/668500b1d20944abb2fec3df69cfcb96?v=67fe2acb766e4e60a04eb702c908efde&p=c2c003628bdd431d946acf5cb7d053f4&pm=s)
